packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -66,25 +66,37 @@ const decorator = (service) => ({
       return service.update.call(this, uid, entityId, { ...opts, data });
     }
 
+    const newStage = await strapi.entityService.findOne(
+      STAGE_MODEL_UID,
+      data[ENTITY_STAGE_ATTRIBUTE]
+    );
     const previousStage = await getEntityStage(uid, entityId);
 
     const updatedEntity = await service.update.call(this, uid, entityId, { ...opts, data });
-    if (
-      previousStage?.workflow?.id &&
-      previousStage?.id &&
-      previousStage.id !== data[ENTITY_STAGE_ATTRIBUTE]
-    ) {
-      const webhookPayload = {
-        entityId,
+
+    if (previousStage?.id && previousStage.id !== newStage.id) {
+      const model = strapi.getModel(uid);
+
+      strapi.eventHub.emit(WORKFLOW_UPDATE_STAGE, {
+        model: model.modelName,
+        uid: model.uid,
+        entity: {
+          id: entityId,
+        },
         workflow: {
           id: previousStage.workflow.id,
           stages: {
-            from: previousStage.id,
-            to: data[ENTITY_STAGE_ATTRIBUTE],
+            from: {
+              id: previousStage.id,
+              name: previousStage.name,
+            },
+            to: {
+              id: newStage.id,
+              name: newStage.name,
```

Comment:
don't you get the stage information inside the updatedEntity? I imagine we populate the entity stage so maybe you can get the new name & id there.